smsjwplatform/jwplatform.py

```diff
@@ -198,6 +211,80 @@ def from_media_id(cls, media_id, preferred_media_type='video'):
         return video_resource
 
 
+class DeliveryVideo(Resource):
+    """
+    A dict subclass representing a video resource object as returned by the Delivery API.
+
+    This subclass provides some convenience accessors for various common resource keys but, since
+    this is a dict subclass, the values can be retrieved using ``[]`` or ``get`` as per usual.
+
+    """
+    @property
+    def media_id(self):
+        """
+        The legacy SMS media id (or None if there is none)
+
+        """
+        field = self.get('sms_media_id')
+        if field is None:
+            return None
+        return parse_custom_field('media', field)
+
+    @property
+    def acl(self):
+        """
+        The parsed ACL custom prop on the resource. If no ACL is present, the WORLD ACL is assumed.
+
+        """
+        field = parse_custom_field('acl', self.get('sms_acl', 'acl:WORLD:'))
+
+        # Work around odd ACL entries. See uisautomation/sms2jwplayer#30.
+        if field == "['']":
+            return []
+
+        return [acl.strip() for acl in field.split(',') if acl.strip() != '']
+
+    @classmethod
+    def from_key(cls, key, client=None):
+        """
+        Return a :py:class:`DeliveryVideo` instance corresponding to the JWPlatform key passed.
+
+        :param key: JWPlatform key for the media.
+        :param client: (optional) an authenticated JWPlatform client as returned by
+            :py:func:`.get_jwplatform_client`. If ``None``, call :py:func:`.get_jwplatform_client`.
+
+        :raises: :py:exc:`VideoNotFoundError` if the video is not found.
+
+        """
+        # Fetch the media download information from JWPlatform.
+        response = DEFAULT_REQUESTS_SESSION.get(
```

Comment:
It strikes me that "default" here means "the only one you can use". Perhaps replace ``client`` in the arguments with ``session`` and have:
```python
session = session if session is not None else DEFAULT_REQUESTS_SESSION
```
at the top of the function and have this line be
```python
response = session.get( ... )
```